[CPDLP-1960] Create service class to send outcomes to TRA

leandroalemao opened the pull request:
### Context

- Ticket: [CPDLP-1960](https://dfedigital.atlassian.net/browse/CPDLP-1960)

### Changes proposed in this pull request

### Guidance to review



[CPDLP-1960]: https://dfedigital.atlassian.net/browse/CPDLP-1960?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

leandroalemao commented:
Still need to add the env variables into the encrypted credentials files. The master.key files from the envs (dev, staging, sandbox & proruction) is required to be able to decrypt the credentials files locally, then once decrypted, I add the new env variables to it and then get the files re-encrypted to be pushed back to github.

cwrw reviewed (commented):
thanks @leandroalemao, a few comments, happy to chat through as well

leandroalemao commented:
thanks @cwrw, I've addressed your comments in https://github.com/DFE-Digital/early-careers-framework/pull/3014/commits/155c675c5d54ab4d810b7fb64dfc8108f18ad009 tks for your review.. 👍 😄

cwrw commented on app/services/qualified_teachers_api_sender.rb:
```diff
@@ -0,0 +1,92 @@
+  validate :not_already_sent
+
+  def call
+    return if invalid?
```
chatted in person, we still need to avoid this as it might fail silently and re-queue, so we might need to move up: `set_sent_to_qualified_teachers_api_at`

cwrw commented on app/services/qualified_teachers_api_sender.rb:
```diff
@@ -0,0 +1,95 @@
+  validates :participant_outcome_id, presence: { message: I18n.t("errors.participant_outcomes.missing_participant_outcome_id") }
+  validates :participant_outcome, presence: { message: I18n.t("errors.participant_outcomes.missing_participant_outcome") }
+  validate :not_already_successfully_sent
+  validate :sent_with_trn_not_found
```
chatted in person: maybe we can look for `already_successfully_sent_to_api` being false instead?

cwrw commented on app/services/qualified_teachers_api_sender.rb:
```diff
@@ -0,0 +1,95 @@
+  end
+
+  def set_qualified_teachers_api_request_successful!
+    participant_outcome.update!(
```
chatted in person to update to `update_column` to avoid callbacks

cwrw commented on app/models/participant_declaration/npq.rb:
```diff
@@ -1,12 +1,22 @@
+    "npq-headship" => "NPQH",
+    "npq-executive-leadership" => "NPQEL",
+    "npq-early-years-leadership" => "NPQEYL",
+    "npq-additional-support-offer" => "NPQASO",
```
could we remove ASO/EHCO as those are not included :)

cwrw commented on app/models/participant_declaration/npq.rb:
```diff
@@ -1,12 +1,22 @@
+    "npq-early-headship-coaching-offer" => "NPQEHCO",
+  }.freeze
 
+  belongs_to :participant_profile, class_name: "ParticipantProfile::NPQ"
```
just curious why we needed this as its in the parent class `ParticipantProfile` 🤔

leandroalemao replied:
Hi @cwrw I don't why I had added this.. but seems all good without it.. tks 👍 😄

cwrw reviewed (approved):
Thanks @leandroalemao, just minor comments but feel free to merge after
